src/main/java/org/jenkinsci/plugins/parallel_test_executor/SplitStep.java

```diff
@@ -62,32 +67,48 @@ public void setStage(String stage) {
         this.stage = stage;
     }
 
-    @Extension public static final class DescriptorImpl extends AbstractStepDescriptorImpl {
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new Execution(context, this);
+    }
 
-        public DescriptorImpl() {
-            super(Execution.class);
+    @Extension
+    public static final class DescriptorImpl extends StepDescriptor {
+
+        @Override
+        public Set<? extends Class<?>> getRequiredContext() {
+            return ImmutableSet.of(TaskListener.class, Run.class);
```

Comment:
Note to self: deliberately omitting `FilePath`, since this is optional.

Reply:
Exactly. I first had `FilePath` here too. Then tests have been failing and I found in the code that there are `@CheckForNull` annotations.